### DISCUSSION TITLE
How to handle errors in Compartments?

### DISCUSSION THREAD

**Discussion by GulgDev** [endojs/endo#1807] [Q&A]
When the module source code has syntax errors, the original error doesn't throw. Instead I can catch error wrapper: `Failed to load module "<module>" in package "<unknown>" ...`. How can I handle the original error and pass it to custom log function?

**Comment by kriskowal** [comment]
@GulgDev Can I trouble you with steps to reproduce? I tried altering the test in `packages/ses/test/test-import-stack-trace.js` and got this kind of error:

```
    ℹ TypeError: Failed to load module "./main.js" in package "<unknown>" (1 underlying failures: Error transforming source in "https://example.com/packages/erroneous/main.js": Missing semicolon. (2:7)
          at load (file:///Users/kris/endo/packages/ses/src/module-load.js:352:11)
          at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
          at async file:///Users/kris/endo/packages/ses/test/test-import-stack-traces.js:69:5
```

I find the underlying transform error (coming from `StaticModuleRecord`) informative. I am wondering what trouble you ran into.
